tests/static-volume_test.go

```diff
@@ -0,0 +1,153 @@
+			dv, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dvDef)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for source DV to succeed")
+			f.ForceBindPvcIfDvIsWaitForFirstConsumer(dv)
+			err = utils.WaitForDataVolumePhaseWithTimeout(f, f.Namespace.Name, cdiv1.Succeeded, dv.Name, 300*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.TODO(), dv.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			pvName = pvc.Spec.VolumeName
+			pvStorageClass = pvc.Spec.StorageClassName
+			pvMode = pvc.Spec.VolumeMode
+
+			By("Retaining PV")
+			pv, err := f.K8sClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			pv.Spec.PersistentVolumeReclaimPolicy = corev1.PersistentVolumeReclaimRetain
+			_, err = f.K8sClient.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Deleting source DV")
+			err = utils.DeleteDataVolume(f.CdiClient, dv.Namespace, dv.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Making PV available")
+			Eventually(func() bool {
+				pv, err := f.K8sClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(pv.Spec.ClaimRef.Namespace).To(Equal(dv.Namespace))
+				Expect(pv.Spec.ClaimRef.Name).To(Equal(dv.Name))
+				if pv.Status.Phase == corev1.VolumeAvailable {
+					return true
+				}
+				pv.Spec.ClaimRef.ResourceVersion = ""
+				pv.Spec.ClaimRef.UID = ""
+				_, err = f.K8sClient.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return false
+			}, timeout, pollingInterval).Should(BeTrue())
+		})
+
+		AfterEach(func() {
+			if pvName == "" {
+				return
+			}
+			err := f.K8sClient.CoreV1().PersistentVolumes().Delete(context.TODO(), pvName, metav1.DeleteOptions{})
+			if errors.IsNotFound(err) {
+				return
+			}
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		DescribeTable("should handle static allocated DataVolume", func(defFunc func() *cdiv1.DataVolume) {
+			By("Creating target DV")
+			dvDef := defFunc()
+			controller.AddAnnotation(dvDef, controller.AnnCheckStaticVolume, "")
+			controller.AddAnnotation(dvDef, controller.AnnDeleteAfterCompletion, "false")
+			dv, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dvDef)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for target DV to succeed")
+			err = utils.WaitForDataVolumePhaseWithTimeout(f, f.Namespace.Name, cdiv1.Succeeded, dv.Name, 300*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.TODO(), dv.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pvc.Spec.VolumeName).To(Equal(pvName))
```

Comment:
This test is great, just one consideration, maybe we should make sure pv.CreationTimestamp < dv.CreationTimestamp

Reply:
sure

Reply:
I am a broken record about this by now, but, maybe we would sleep better if there is an md5 check at the end of this test

Reply:
not a bad idea